Altimeter/Inc/altimeter.hpp

```diff
@@ -0,0 +1,166 @@
+#ifndef INC_ALTIMETER_HPP_
+#define INC_ALTIMETER_HPP_
+
+#include "stm32l5xx_hal.h"
+#include "cmsis_os.h"
+#include <cstdint>
+#include <cmath>
+
+
+#define RESET 0x1E
+
+/*PROM Addresses */
+#define PROM_READ_ADDRESS_0 0xA0
+#define PROM_READ_ADDRESS_1 0xA2
+#define PROM_READ_ADDRESS_2 0xA4
+#define PROM_READ_ADDRESS_3 0xA6
+#define PROM_READ_ADDRESS_4 0xA8
+#define PROM_READ_ADDRESS_5 0xAA
+#define PROM_READ_ADDRESS_6 0xAC
+#define PROM_READ_ADDRESS_7 0xAE
+
+
+/*digital pressure/temperature commands at different sampling rates*/
+#define CONVERT_D1_OSR_256 0x40
+#define CONVERT_D1_OSR_512 0x42
+#define CONVERT_D1_OSR_1024 0x44
+#define CONVERT_D1_OSR_2048 0x46
+#define CONVERT_D1_OSR_4096 0x48
+#define CONVERT_D2_OSR_256 0x50
+#define CONVERT_D2_OSR_512 0x52
+#define CONVERT_D2_OSR_1024 0x54
+#define CONVERT_D2_OSR_2048 0x56
+#define CONVERT_D2_OSR_4096 0x58
+
+#define ADC_READ 0x0
+
+#define TIMEOUT 1 << 31
+
+typedef struct cal_coeffs{
+	uint16_t sens;
+	uint16_t off;
+	uint16_t tcs;
+	uint16_t tco;
+	uint16_t t_ref;
+	uint16_t temp_sens;
+}cal_coeffs;
+
+
+
+/* To get temperature/pressure/altitude values, call (step 1) AltDevice, then (step 2) AltInit once.
+ * Then before getting (step 4) pressure/temp/altitude, (step 3) calculateTempPres should be called beforehand.
+ *  */
+
+class AltDevice{
+public:
+	/**
+	* Constructor for the AltDevice class
+	 *
+	* @param ps_port -> protocol select port
+	* @param ps_pin -> protocol select pin
+	 *
+	* @return none
+	 */
+	AltDevice(GPIO_TypeDef *ps_port, uint16_t ps_pin);
```

Comment:
In my opinion, a driver code could be a lot easier to use if we only initiate the device once with the SPI, cs, ps line bundled with this driver code so that the user of your driver doesn't have to know the SPI it is using every time they call get_altitude function. That means I think `SPI_HandleTypeDef` and `GPIO_TypeDef` they should be the private variable of this class.